docs/production-deployment/worker-deployments/worker-versioning.mdx

```diff
@@ -395,6 +395,45 @@ temporal workflow reset with-workflow-update-options \
     --versioning-override-build-id "$TARGET_BUILD_ID"
 ```
 
+## Migrating a Workflow from Pinned to Auto-Upgrade
+
+There may be times when you need to migrate your Workflow from Pinned to Auto-Upgrade because you configured your Workflow Type with the wrong behavior or you've pinned a really long-running Workflow by mistake.
+
+Pinned Workflows get stuck on versions and Workflows prevent versions from draining, long-running Workflows could be preventing drainage. You could move the Workflow to a new build, but that would just push the problem to the next build.
+
+In order to make this change, you need to change the versioning behavior for your Workflow from Pinned to Auto-Upgrade. You can use `temporal workflow update-options` for this:
+
+```bash
+temporal workflow update-options \
+    --workflow-id "$WORKFLOW_ID" \
+    --versioning-override-behavior auto_upgrade \
+    --versioning-override-deployment-name "$TARGET_DEPLOYMENT"
+```
+
+If you want to move all of your Workflows of a certain type to this new configuration, you can do it with this command:
+
+```bash
+temporal workflow update-options \
+    --query="WorkflowType="$WORKFLOW_TYPE" \
+    --workflow-id "$WORKFLOW_ID" \
+    --versioning-override-behavior auto_upgrade \
+    --versioning-override-deployment-name "$TARGET_DEPLOYMENT"
+```
+
+You can also filter on a certain build ID to limit the number of Workflows you apply it to:
+
+```bash
+temporal workflow update-options \
+    --query "WorkflowType="$WORKFLOW_TYPE" AND --TemporalWorkerDeploymentVersion=$TARGET_DEPLOYMENT:$OLD_VERSION" \
+    --workflow-id "$WORKFLOW_ID" \
+    --versioning-override-behavior auto_upgrade \
+    --versioning-override-deployment-name "$TARGET_DEPLOYMENT"
+```
+
+:::note
```

Comment:
Maybe we want to clarify the following point?

Upon changing the behavior to auto-upgrade, the workflow will resume work on the Current version. So if the Current version is different from the previous pinned version of the workflow, they should be compatible (i.e. patching should have been use as needed). Otherwise, the workflow will be stuck on the Current version.